neural_compressor/common/base_tuning.py

```diff
@@ -0,0 +1,270 @@
+        return overall_result + eval_result * eval_pair[self.WEIGHT]
+
+    def get_number_of_eval_functions(self) -> int:
+        return len(self.eval_fn_registry)
+
+    def _set_eval_fn_registry(self, user_eval_fns: List[Dict]) -> None:
+        self.eval_fn_registry = [
+            {
+                self.EVAL_FN: user_eval_fn_pair[self.EVAL_FN],
+                self.WEIGHT: user_eval_fn_pair.get(self.WEIGHT, 1.0),
+                self.FN_NAME: user_eval_fn_pair.get(self.FN_NAME, user_eval_fn_pair[self.EVAL_FN].__name__),
+            }
+            for user_eval_fn_pair in user_eval_fns
+        ]
+
+    def set_eval_fn_registry(self, eval_fns: Optional[Union[Callable, Dict, List[Dict]]] = None) -> None:
+        # About the eval_fns format, refer the class docstring for details.
+        if eval_fns is None:
+            return
+        elif callable(eval_fns):
+            # single eval_fn
+            eval_fn_pair = copy.deepcopy(self.EVAL_FN_TEMPLATE)
+            eval_fn_pair[self.EVAL_FN] = eval_fns
+            eval_fn_pair[self.FN_NAME] = eval_fns.__name__
+            eval_fns = [eval_fn_pair]
+        elif isinstance(eval_fns, Dict):
+            eval_fns = [eval_fns]
+        elif isinstance(eval_fns, List):
+            assert all([isinstance(eval_fn_pair, Dict) for eval_fn_pair in eval_fns])
+        else:
+            raise NotImplementedError(f"The eval_fns should be a dict or a list of dict, but got {type(eval_fns)}.")
+        self._set_eval_fn_registry(eval_fns)
+
+    def self_check(self) -> None:
+        # check the number of evaluation functions
+        num_eval_fns = self.get_number_of_eval_functions()
+        assert num_eval_fns > 0, "Please ensure that you register at least one evaluation metric for auto-tune."
+        logger.info("There are %d evaluations functions.", num_eval_fns)
+
+
+evaluator = Evaluator()
+
+
+class Sampler:
+    pass
+
+
+class ConfigLoader:
+    def __init__(self, quant_configs, sampler: Sampler) -> None:
+        self.quant_configs = quant_configs
+        self.sampler = sampler
+
+    @staticmethod
+    def parse_quant_config(quant_config: BaseConfig) -> List[BaseConfig]:
+        if isinstance(quant_config, ComposableConfig):
+            result = []
+            for q_config in quant_config.config_list:
+                result += q_config.expand()
+            return result
+        else:
+            return quant_config.expand()
+
+    def parse_quant_configs(self) -> List[BaseConfig]:
+        # TODO (Yi) separate this functionality into `Sampler` in the next PR
+        quant_config_list = []
+        for quant_config in self.quant_configs:
+            quant_config_list.extend(ConfigLoader.parse_quant_config(quant_config))
+        return quant_config_list
+
+    def __iter__(self) -> Generator[BaseConfig, Any, None]:
+        for config in self.parse_quant_configs():
+            yield config
+
+
+class TuningLogger:
```

Comment:
Hi @chensuyue, here is a unified logger used in `autotune`.  I hope it can help you retrieve the log. We can refine the log content when needed.

Reply:
Thanks, that's good enough for now.